zipline/gens/realtimeclock.py

```diff
@@ -49,18 +49,34 @@ def __init__(self,
                  minute_emission,
                  time_skew=pd.Timedelta("0s"),
                  is_broker_alive=None):
-        self.sessions = sessions
-        self.execution_opens = execution_opens
-        self.execution_closes = execution_closes
-        self.before_trading_start_minutes = before_trading_start_minutes
+        today = pd.to_datetime('now', utc=True).date()
+        beginning_of_today = pd.to_datetime(today, utc=True)
+        beginning_of_tomorrow = beginning_of_today + pd.Timedelta('1 day')
+        self.sessions = \
+            sessions[(beginning_of_today <= sessions) &
+                     (sessions < beginning_of_tomorrow)]
+        self.execution_opens = \
+            execution_opens[(beginning_of_today <= execution_opens) &
+                            (execution_opens < beginning_of_tomorrow)]
+        self.execution_closes = \
+            execution_closes[(beginning_of_today <= execution_closes) &
+                             (execution_closes < beginning_of_tomorrow)]
+        self.before_trading_start_minutes = \
+            before_trading_start_minutes[
+                (beginning_of_today <= before_trading_start_minutes) &
+                (before_trading_start_minutes < beginning_of_tomorrow)]
+
         self.minute_emission = minute_emission
         self.time_skew = time_skew
         self.is_broker_alive = is_broker_alive or (lambda: True)
         self._last_emit = None
         self._before_trading_start_bar_yielded = False
 
     def __iter__(self):
-        yield self.sessions[0], SESSION_START
+        if not len(self.sessions):
```

Comment:
in what case would it end up with no sessions?

Reply:
If the algo's very first start was done at a non-trading day (e.g. weekend). 
I tend to bootstrap my algos over the weekend, hence caught this bug.